File: src/main/webapp/resources/js/page/pageMaker.js
$('head').append('<script src=\'../../../resources/js/page/criteria.js\'><\/script>');

function PageMaker() {

	  this.totalCount; //전체 글개수
	  this.startPage;  //시작 페이지
	  this.endPage;    //끝 페이지
	  this.prev;       //boolean타입
	  this.next;       //boolean타입
	  this.displayPageNum;
	  this.page;	
	  this.perPageNum;
	  
//	  this.displayPageNum = 10; //보여줄 페이지번호 개수
	   
//	   this.cri = new Criteria();
//
//	   this.setTotalCount = function(totalCount) { //전체 개수를 정해준다.
//	      this.totalCount = totalCount;
//	      
//	      calcData();
//	   };

//	   this.calcData = function() { //페이지 계산
//	      
//	      endPage = (Math.ceil(cri.page() / displayPageNum ) * displayPageNum);
//	      
//	      startPage = (endPage - displayPageNum) + 1;
//	      
//	      var tempEndPage = (Math.ceil(totalCount / cri.getPerPageNum()));
//	      
//	      if(endPage > tempEndPage){
//	         endPage = tempEndPage;
//	      }
//	      
//	      prev = startPage ==1 ? false : true; //startPage가 1이면 false 아니면 true
//	      
//	      next = endPage * cri.getPerPageNum() >= totalCount ? false : true;
//	      
//	   };

//	   this.makeQuery = function(page){
//	      
//	      UriComponents uriComponents =
//	               UriComponentsBuilder.newInstance()
//	               .queryParam("page", page)
//	               .queryParam("perPageNum", cri.getPerPageNum())
//	               .build();               
//	      
//	      return uriComponents.toUriString();
//	      
//
//
//	   }
//
//	   public String makeSearch(int page){
//	      
//	      UriComponents uriComponents =
//	               UriComponentsBuilder.newInstance()
//	               .queryParam("page", page)
//	               .queryParam("perPageNum", cri.getPerPageNum())
//	               .queryParam("searchType", ((SearchCriteria)cri).getSearchType())
//	               .queryParam("keyword", ((SearchCriteria)cri).getKeyword())
//	               .build();               
//	      
//	      return uriComponents.toUriString();
//	   }   
	
}

File: src/main/java/com/phoenix/hiddenplace/domain/PageMaker.java
package com.phoenix.hiddenplace.domain;

import org.springframework.web.util.UriComponents;
import org.springframework.web.util.UriComponentsBuilder;


public class PageMaker {

	private int totalCount;
	private int startPage;
	private int endPage;
	private boolean prev;
	private boolean next;
	private int page = 1;	
	private int perPageNum = 12;
	private int themeCode = 0;
	private int displayPageNum = 10;
	
	public PageMaker() {
		
	}
//	private Criteria cri;
//
//
//	public void setCri(Criteria cri) {
//		this.cri = cri;
//	}

	public void setTotalCount(int totalCount) {
		this.totalCount = totalCount;
		
		this.calcData();
	}

	private void calcData() {
		
		this.endPage = (int)(Math.ceil(getPage() / (double)displayPageNum ) * displayPageNum);

		this.startPage = (endPage - displayPageNum) + 1;
		
		int tempEndPage = (int)(Math.ceil(totalCount / (double)getPerPageNum()));
		
		if(endPage > tempEndPage){
			this.endPage = tempEndPage;
		}

		this.prev = startPage ==1 ? false : true;
		
		this.next = endPage * getPerPageNum() >= totalCount ? false : true;
		
	}

	public int getTotalCount() {
		return totalCount;
	}

	public int getStartPage() {
		return startPage;
	}

	public int getEndPage() {
		return endPage;
	}

	public boolean isPrev() {
		return prev;
	}

	public boolean isNext() {
		return next;
	}

	public int getDisplayPageNum() {
		return displayPageNum;
	}

	public String makeQuery(int page){
		
		UriComponents uriComponents =
	            UriComponentsBuilder.newInstance()
	            .queryParam("page", page)
	            .queryParam("perPageNum", getPerPageNum())
	            .build();	            
		
		return uriComponents.toUriString();
	}


	public void setStartPage(int startPage) {
		this.startPage = startPage;
	}

	public void setEndPage(int endPage) {
		this.endPage = endPage;
	}

	public void setPrev(boolean prev) {
		this.prev = prev;
	}

	public void setNext(boolean next) {
		this.next = next;
	}

	public void setDisplayPageNum(int displayPageNum) {
		this.displayPageNum = displayPageNum;
	}	
	
	public void setPage(int page){

		if(page <= 0){
			this.page = 1;
			return;
		}

		this.page = page;
	}

	public void setPerPageNum(int perPageNum){

		if(perPageNum <= 0 || perPageNum > 100){
			this.perPageNum = 12;
			return;
		}

		this.perPageNum = perPageNum;
	}

	public int getPage() {
		return page;
	}	

	//method for MyBatis SQL Mapper - 
	public int getPageStart() {

		return (this.page -1)* perPageNum;
		
	}

	//method for MyBatis SQL Mapper 
	public int getPerPageNum(){

		return this.perPageNum;
	}
	
	public int getThemeCode() {
		return themeCode;
	}

	public void setThemeCode(int themeCode) {
		this.themeCode = themeCode;
	}

	
}